### ISSUE TITLE
`power.deep_sleep()` without wake up sources does not go to sleep

### ISSUE THREAD

**Issue by microbit-carlos** [microbit-foundation/micropython-microbit-v2#126]
This might be related to this CODAL issue:
- https://github.com/lancaster-university/codal-microbit-v2/issues/203

However, based on the latest conversations with @JohnVidler we were thinking the issue might be in the MakeCode extension instead of CODAL.

However, in MicroPython executing `power.deep_sleep()` doesn't go to sleep either. Having a brief look the MicroPython code I can't see an obvious reason why it might not go to sleep if it's supported within CODAL itself:
https://github.com/microbit-foundation/micropython-microbit-v2/blob/2233d79b8ab15bb769adaa44d53874ce80ebd3eb/src/codal_port/modpower.c#L54-L109
https://github.com/microbit-foundation/micropython-microbit-v2/blob/6ea476f3ddb4eb7aba13ca264fab9d15abaae3ca/src/codal_app/microbithal.cpp#L130-L136

Example programme to replicate with MicroPython v2.1.0-beta.3:
```python
from microbit import *
import power

while True:
    power.deep_sleep()
    display.show(Image.SURPRISED)
    sleep(500)
```
This always shows the SURPRISED face and doesn't go to sleep

**Comment by dpgeorge** [comment]
I think this is a "feature" of the CODAL: if there are no wake up sources configured, and no timeout, then the `deepSleep()` call returns immediately (presumably to prevent infinite waits that cannot be interrupted, requiring a reflash of the microbit to recover).

**Comment by dpgeorge** [comment]
We could fix this on the MicroPython side by just calling the CODAL `deepSleep(ms, true)` with a large timeout, and keep calling it in a loop forever.

**Comment by microbit-carlos** [comment]
This has been changed/fixed in the CODAL side in this PR, so we can just wait for the next tag to get the fix:
- https://github.com/lancaster-university/codal-microbit-v2/pull/229

**Comment by microbit-carlos** [comment]
This is resolved in CODAL v0.2.43, we can close it when https://github.com/microbit-foundation/micropython-microbit-v2/pull/132 is merged, or a different commit updates CODAL in the main branch.

**Comment by dpgeorge** [comment]
CODAL v0.2.43 is updated here.
